extensions/VisualEditor/wikia/modules/ve/ui/pages/ve.ui.WikiaParameterPage.js

```diff
@@ -6,6 +6,104 @@
  */
 
 /**
+ * Restores infobox dialog info when going back from media dialog
+ *
+ * @param {ve.ui.WindowManager} win
+ * @param {ve.dm.WikiaTransclusionModel} model - original tranclusion model of infobox dialog
+ * @param {string} imageName - image name from media dialog
+ * @param {string} paramName - Name of a field (e.g. image1)
+ */
+function restoreInfoboxDialog(win, model, imageName, paramName) {
+	if (imageName) {
+		model.parts[0].params[paramName].value = imageName;
+	}
+
+	win.transclusionModel = model;
+
+	win.bookletLayout.$element.html('');
+	win.initializeTemplateParameters();
+}
+
+/**
+ * shorthand function to open dialogs
+ *
+ * @param {string} name - Name of a dialog to open
+ */
+
+function openDialog(name) {
+	ve.ui.actionFactory
+		.create('window', ve.init.target.getSurface())
+		.open(name);
+}
+/**
+ *
+ * @param {ve.ui} currentWindow
+ * @param {ve.dm.WikiaCart} cartItems
+ * @returns {string|null}
+ */
+function getImageName(currentWindow, cartItems) {
+	if (
+		currentWindow.currentAction &&
+		currentWindow.currentAction.getAction() === 'goback' &&
+		cartItems.length > 0
+	) {
+		return cartItems[0].data;
+	}
+
+	return null;
+}
+/**
+ * In order to intercept default behaviour of wikiaMediaInsert
+ * additional action was added to the dialog
+ * goback - it does nothing, but allows us to get images chosen in the dialog
+ *
+ * @param {string} action action name (e.g. goback, apply)
+ */
+function setDefaultMediaInsertDialogAction(action) {
+	ve.ui.WikiaMediaInsertDialog.static.actions[0].action = action;
+}
+
+/**
+ * Event handler on image button
+ * Opens wikiaMediaInsert dialog and allows user to choose an image
+ * from wiki images
+ *
+ * @param {string} paramName Name of a field (e.g. image1)
+ */
+function handleClickOnImageButton(paramName) {
+	var windowManager = ve.init.target.getSurface().getDialogs(),
+		transclusionModel = windowManager.currentWindow.transclusionModel;
+
+	windowManager.closeWindow('wikiaInfobox').done(function () {
+		openDialog('wikiaMediaInsert');
+		setDefaultMediaInsertDialogAction('goback');
+
+		windowManager.once('closing', function (currentWindow) {
+			var imageName;
+
+			if (currentWindow instanceof ve.ui.WikiaMediaInsertDialog) {
```

Comment:
in theory this should not happen, but
I assign a handler to the windowManager so technically any dialog could open and this would fire